machine-runner/src/runner/runner.ts

```diff
@@ -46,9 +48,13 @@ import { MachineEmitter, MachineEmitterEventMap } from './runner-utils.js'
  * }
  * machine.isDestroyed() // returns true
  */
-export type MachineRunner = {
+export type MachineRunner<
+  SwarmProtocolName extends string,
+  MachineName extends string,
+  RegisteredEventsFactoriesTuple extends MachineEvent.Factory.Any[],
```

Comment:
Why is this parameter needed?

Reply:
Ah, it was because I copy-pasted the generics down to StateOpaque. Partly that,

Partly because I figure we might need `StateOpaque.protocolName()` and `StateOpaque.machineName()` for display purposes so it might be needed there.

But I have no strong reason for this to be here in the moment.

Reply:
Adding type parameters incurs a cost that I consider to be rather high — it makes all code using this type more complex. This means that every type parameter needs a strong explanation for why it exists — and it needs to be stronger the more parameters there are.

Reply:
I see. I'll minimize

Reply:
I found one functionality it can be useful for:
- It is used to limit the possible StateFactory parameter of `StateOpaque.is` and `StateOpaque.as`.
- It is useful to reduce mistakes due to both SwarmName and MachineName being a literal strings.
- It is an edge case that can also be handled at runtime.

Reply:
`symbol` only works at runtime as far as I remember.
`unique symbol` is not supported for instantiated objects.